packages/components/addon/components/hds/code-block/index.js

```diff
@@ -0,0 +1,154 @@
+import { htmlSafe } from '@ember/template';
+import { guidFor } from '@ember/object/internals';
+
+import Prism from 'prismjs';
+
+import 'prismjs/plugins/line-numbers/prism-line-numbers';
+import 'prismjs/plugins/line-highlight/prism-line-highlight';
+
+import 'prismjs/components/prism-bash';
+import 'prismjs/components/prism-go';
+import 'prismjs/components/prism-markup-templating';
+import 'prismjs/components/prism-handlebars';
+import 'prismjs/components/prism-hcl';
+import 'prismjs/components/prism-json';
+import 'prismjs/components/prism-log';
+import 'prismjs/components/prism-shell-session';
+import 'prismjs/components/prism-yaml';
+
+export default class HdsCodeBlockIndexComponent extends Component {
+  @tracked prismCode = '';
+
+  /**
+   * Generates a unique ID for the code content
+   *
+   * @param preCodeId
+   */
+  preCodeId = 'pre-code-' + guidFor(this);
+
+  /**
+   * @param code
+   * @type {string}
+   * @description code text content for the CodeBlock
+   */
+  get code() {
+    const code = this.args.value;
+
+    assert(
+      '@code for "Hds::CodeBlock" must have a valid value',
+      code !== undefined
+    );
+
+    if (Prism?.plugins?.NormalizeWhitespace) {
+      return Prism.plugins.NormalizeWhitespace.normalize(code);
+    }
+
+    return code;
+  }
+
+  /**
+   * @param language
+   * @type {string}
+   * @default undefined
+   * @description name of coding language used within CodeBlock for syntax highlighting
+   */
+  get language() {
+    return this.args.language ?? undefined;
+  }
+
+  /**
+   * @param hasLineNumbers
+   * @type {boolean}
+   * @default true
+   * @description Displays line numbers if true
+   */
+  get hasLineNumbers() {
+    return this.args.hasLineNumbers ?? true;
+  }
+
+  /**
+   * @param isStandalone
+   * @type {boolean}
+   * @default true
+   * @description Make CodeBlock container corners appear rounded
+   */
+  get isStandalone() {
+    return this.args.isStandalone ?? true;
+  }
+
+  /**
+   * @param hasLineWrapping
+   * @type {boolean}
+   * @default false
+   * @description Make text content wrap on multiple lines
+   */
+  get hasLineWrapping() {
+    return this.args.hasLineWrapping ?? false;
+  }
+
+  @action
+  setPrismCode(element) {
+    const code = this.code;
+    const language = this.language;
+    const grammar = Prism.languages[language];
+
+    if (code && language && grammar) {
+      this.prismCode = htmlSafe(Prism.highlight(code, grammar, language));
+    } else {
+      this.prismCode = htmlSafe(Prism.util.encode(code));
+    }
+
+    // Force prism-line-numbers plugin initialization, required for Prism.highlight usage
+    // See https://github.com/PrismJS/prism/issues/1234
+    Prism.hooks.run('complete', {
+      code,
+      element,
+    });
+
+    // Force prism-line-highlight plugin initialization
+    if (this.args.highlightLines) {
+      setTimeout(() => {
+        if (window) window.dispatchEvent(new Event('resize'));
+      }, 100);
+    }
```

Comment:
This is certainly not my proudest piece of code but was the only way to make the line-highlight plugin work in all instances [as flagged in this comment](https://github.com/hashicorp/design-system/pull/1749#discussion_r1372296664).

Reply:
Good solution I think in this case. I kept trying different delays but couldn't figure out anything that would work myself.